services/requests/request_definitions/models.go

```diff
@@ -0,0 +1,47 @@
+package request_definitions
+
+import "github.com/leanspace/terraform-provider-leanspace/helper/general_objects"
+
+type RequestDefinition struct {
+	ID                               string                            `json:"id"`
+	Name                             string                            `json:"name"`
+	Description                      string                            `json:"description,omitempty"`
+	PlanTemplateIds                  []string                          `json:"planTemplateIds"`
+	FeasibilityConstraintDefinitions []FeasibilityConstraintDefinition `json:"feasibilityConstraintDefinitions"`
+	ConfigurationArgumentDefinitions []ArgumentDefinition[any]         `json:"configurationArgumentDefinitions"`
+	ConfigurationArgumentMappings    []ArgumentMapping                 `json:"configurationArgumentMappings"`
+	CreatedAt                        string                            `json:"createdAt"`
+	CreatedBy                        string                            `json:"createdBy"`
+	LastModifiedAt                   string                            `json:"lastModifiedAt"`
+	LastModifiedBy                   string                            `json:"lastModifiedBy"`
+}
+
+func (requestDefinition *RequestDefinition) GetID() string {
+	return requestDefinition.ID
+}
+
+type FeasibilityConstraintDefinition struct {
+	ID                  string                    `json:"id"`
+	Name                string                    `json:"name"`
+	Description         string                    `json:"description,omitempty"`
+	Required            bool                      `json:"required,omitempty"`
+	Cloned              bool                      `json:"cloned"`
+	ArgumentDefinitions []ArgumentDefinition[any] `json:"argumentDefinitions,omitempty"`
+	CreatedAt           string                    `json:"createdAt"`
+	CreatedBy           string                    `json:"createdBy"`
+	LastModifiedAt      string                    `json:"lastModifiedAt"`
+	LastModifiedBy      string                    `json:"lastModifiedBy"`
+}
+
+type ArgumentDefinition[T any] struct {
+	Name        string                                 `json:"name"`
+	Description string                                 `json:"description,omitempty"`
+	Attributes  general_objects.DefinitionAttribute[T] `json:"attributes"`
+}
```

Comment:
You can reuse the type defined in feasibility_constraint_definitions to avoid duplication

Reply:
What is the rule about it ?   
I saw that many resources re-defined Metadata for example (streams, widgets, ...) so I thought duplication was expected to avoid package dependencies.   
Also, this `ArgumentDefinition` has been first created in the `activity_definitions` package so should I re-use it ?

We may want : 
- no dependency between packages : what I have done
- dependencies allowed between packages of the same API/resource type : I duplicate the ArgumentDefinition in the feasibility_constraint_definition and call it from request_definitions
-  dependencies allowed : both can import activity_definitions ArgumentDefinition
which one should we apply ?

Reply:
Ideally I don't think we should have duplication, if resource shared some common schemas it can reused. To make it clearer they may be moved to another file/package (i.e. helper) so that we don't have to think where it comes from originally
So I would go with the last option, I suspect the duplication is because multiple people worked on it and we didn't pay attention to this part

Reply:
applied